ludwig/models/ecd.py

```diff
@@ -271,6 +278,14 @@ def recurse_weights(model, prefix=None):
 
         return weights
 
+    def __setstate__(self, newstate):
```

Comment:
trying to get a better understanding of __setstate__ and __reduce__. Why are they needed? How are they used?